### PR TITLE
Parse .sav files

### PR DESCRIPTION
**Pull request by tgjones**

### CODE REVIEW

**Comment by tgjones** [comment]
There’s a lot more to do here but since it’s all new stuff, no point holding it up.

File: src/OpenSage.Game/Data/Sav/SaveFile.cs [hunk 1]
```diff
@@ -1,25 +1,505 @@
...
+
+        public static SaveFile FromStream(Stream stream)
         {
-            using (var stream = entry.Open())
             using (var reader = new BinaryReader(stream, Encoding.Unicode, true))
             {
-                var chunkName = reader.ReadBytePrefixedAsciiString();
+                var chunkHeaders = new List<SaveChunkHeader>();
+
+                while (true)
```
Comment (Tarcontar) [inline src/OpenSage.Game/Data/Sav/SaveFile.cs]: maybe use chunkHeader.IsEof or sth else here? just has some code smell to me

File: src/OpenSage.Game/Data/Sav/SaveFile.cs [hunk 2]
```diff
@@ -1,25 +1,505 @@
...
+
+                if (result.Unknown1 != 1u)
+                {
+                    throw new InvalidDataException();
+                }
+
+                if (result.Unknown2 != 1u)
+                {
+                    //throw new InvalidDataException();
+                }
```
Comment (Tarcontar) [inline src/OpenSage.Game/Data/Sav/SaveFile.cs]: whole if statement could be removed, just add a comment to Unknown2

**Comment by feliwir** [comment]
What's blocking this from getting merged?

**Comment by tgjones** [comment]
Time to make the fixes that @Tarcontar suggested.

**Pull request merged**